libs/product/src/drivers/magento/transforms/configurable-product-transformer.spec.ts

```diff
@@ -0,0 +1,192 @@
+							label: daffConfigurableProduct.configurableAttributes[0].values[2].label,
+							swatch_data: {
+								value: daffConfigurableProduct.configurableAttributes[0].values[2].swatch.value,
+								thumbnail: null
+							}
+						}
+					],
+					attribute_id: null,
+					product_id: null,
+					id: null
+				}
+			],
+			variants: [
+				{
+					attributes: [
+						{
+							code: daffConfigurableProduct.configurableAttributes[0].code,
+							label: daffConfigurableProduct.configurableAttributes[0].label,
+							value_index: parseInt(daffConfigurableProduct.configurableAttributes[0].values[0].value, 10)
+						}
+					],
+					product: {
+						image: {
+							url: daffConfigurableProduct.variants[0].image.url,
+							label: daffConfigurableProduct.variants[0].image.label
+						},
+						price_range: {
+							maximum_price: {
+								regular_price: {
+									value: parseInt(daffConfigurableProduct.variants[0].price, 10),
+									currency: null
+								}
+							}
+						},
+						sku: daffConfigurableProduct.variants[0].id, 
+						id: null, 
+						name: null, 
+						__typename: null,
+						url_key: null, 
+						thumbnail: null
+					}
+				},
+				{
+					attributes: [
+						{
+							code: daffConfigurableProduct.configurableAttributes[0].code,
+							label: daffConfigurableProduct.configurableAttributes[0].label,
+							value_index: parseInt(daffConfigurableProduct.configurableAttributes[0].values[1].value, 10)
+						}
+					],
+					product: {
+						image: {
+							url: daffConfigurableProduct.variants[1].image.url,
+							label: daffConfigurableProduct.variants[1].image.label
+						},
+						price_range: {
+							maximum_price: {
+								regular_price: {
+									value: parseInt(daffConfigurableProduct.variants[1].price, 10),
+									currency: null
+								}
+							}
+						},
+						sku: daffConfigurableProduct.variants[1].id, 
+						id: null, 
+						name: null, 
+						__typename: null,
+						url_key: null, 
+						thumbnail: null
+					}
+				},
+				{
+					attributes: [
+						{
+							code: daffConfigurableProduct.configurableAttributes[0].code,
+							label: daffConfigurableProduct.configurableAttributes[0].label,
+							value_index: parseInt(daffConfigurableProduct.configurableAttributes[0].values[2].value, 10)
+						}
+					],
+					product: {
+						image: {
+							url: daffConfigurableProduct.variants[2].image.url,
+							label: daffConfigurableProduct.variants[2].image.label
+						},
+						price_range: {
+							maximum_price: {
+								regular_price: {
+									value: parseInt(daffConfigurableProduct.variants[2].price, 10),
+									currency: null
+								}
+							}
+						},
+						sku: daffConfigurableProduct.variants[2].id, 
+						id: null, 
+						name: null, 
+						__typename: null,
+						url_key: null, 
+						thumbnail: null
+					}
+				}
+			]
+		}
+  });
+
+	describe('transform', () => {
```

Comment:
I think this spec is rather difficult to understand from a testing perspective, I think my other comment may help alleviate this complexity.

Reply:
Making the factory return other fields won't actually help, because those fields will differ from the ones defined by the DaffCompositeProductFactory. So I'll need to overwrite everything anyway. Transformer tests are just really unfortunate to look at, and I'm not sure what we can really do to make them better.

Reply:
I was able to remove some of the null setting of fields, but I think everything else should stay